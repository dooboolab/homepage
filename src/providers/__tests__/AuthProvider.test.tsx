import * as React from 'react';

import {AuthProvider, useAuthContext} from '../AuthProvider';
import {Button, View} from 'react-native';
import {RenderAPI, act, fireEvent, render} from '@testing-library/react-native';

const FakeChild = (): React.ReactElement => {
  const {setUser} = useAuthContext();

  return (
    <View>
      <Button
        testID="BUTTON"
        onPress={(): void =>
          // @ts-ignore
          setUser({
            displayName: 'displayName',
          })
        }
        title="Button"
      />
    </View>
  );
};

describe('Rendering', () => {
  const component = (
    <AuthProvider>
      <FakeChild />
    </AuthProvider>
  );

  const testingLib: RenderAPI = render(component);

  it('component and snapshot matches', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('Interactions', () => {
  it('should setUser', async () => {
    const {getByTestId} = render(
      <AuthProvider>
        <FakeChild />
      </AuthProvider>,
    );

    act(() => {
      fireEvent.press(getByTestId('BUTTON'));
    });
  });
});
