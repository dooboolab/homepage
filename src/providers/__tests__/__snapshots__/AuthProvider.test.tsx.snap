// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering component and snapshot matches 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    onPress={[MockFunction]}
    testID="BUTTON"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;
